app/shared/components/about-us/our-mission/OurMission.tsx

```diff
@@ -1,166 +1,194 @@
-    missionPoints: DEFAULT_MISSION_POINTS.map((point) => ({
-      id: crypto.randomUUID(),
-      value: point
-    })),
-    imageBlocks: IMAGE_BLOCKS_INITIAL
-  });
-
+  const currentLocale: 'uk' | 'en' = useStore((state) => state.locale);
+  const { block } = usePageBlock(pageId, blockId);
   const setField = useStore((state) => state.setField);
 
-  const handleChangeImage = (id: string, file: File) => {
-    const newUrl = URL.createObjectURL(file);
-    const updated = block.imageBlocks.map((img: MissionImage) =>
-      img.id === id ? { ...img, imageUrl: newUrl, fileName: file.name } : img
-    );
-    setField(pageId, blockId, 'imageBlocks', updated);
-  };
+  if (!block) return <Skeleton sx={{ height: '60px' }} />;
 
-  const handleChangeCaption = (id: string, caption: string) => {
-    const updated = block.imageBlocks.map((img: MissionImage) => (img.id === id ? { ...img, caption } : img));
-    setField(pageId, blockId, 'imageBlocks', updated);
-  };
+  const missionList: MissionListItemWithId[] = ensureIds(block.list);
 
-  const handleChangeMissionPoint = (updatedPoint: MissionPoint) => {
-    const updated = block.missionPoints.map((point: MissionPoint) =>
-      point.id === updatedPoint.id ? updatedPoint : point
+  const missionPoints: MissionPoint[] = missionList.map((item) => ({
+    id: item.id,
+    value: proseToText(item[currentLocale])
+  }));
+
+  const handleChangeMissionPoint = (id: string | number, value: string) => {
+    const updatedList: MissionListItemWithId[] = missionList.map((item) =>
+      item.id === id ? { ...item, [currentLocale]: textToProse(value) } : item
     );
-    setField(pageId, blockId, 'missionPoints', updated);
+    setField(pageId, blockId, 'list', updatedList);
+  };
+  const handleAddMissionPoint = (): MissionPoint => {
+    const newItem: MissionListItemWithId = {
+      uk: { type: 'doc', content: [] },
+      en: { type: 'doc', content: [] },
+      id: crypto.randomUUID()
+    };
+    const newList = [...missionList, newItem];
+    setField(pageId, blockId, 'list', newList);
+    return { id: newItem.id, value: '' };
   };
 
   const handleDeleteMissionPoint = (id: string | number) => {
-    setField(
-      pageId,
-      blockId,
-      'missionPoints',
-      block.missionPoints.filter((point: MissionPoint) => point.id !== id)
-    );
+    const newList = missionList.filter((item) => item.id !== id);
+    setField(pageId, blockId, 'list', newList);
   };
 
-  const handleCreateMissionPoint = () => {
-    const newPoint: MissionPoint = {
-      id: crypto.randomUUID(),
-      value: ''
+  const handleImageChange = (key: 'smallImage' | 'bigImage', value: string, file?: File) => {
+    const image = block[key];
+    if (!image) return;
+
+    const updated = {
+      ...image,
+      caption: { ...image.caption, [currentLocale]: value },
+      ...(file && { src: file.name, generatedSrc: `/api/blob-url?folderName=photos&blobName=${file.name}` })
```

Comment:
maybe it would be better to move this path into constants?
```suggestion
      ...(file && { src: file.name, generatedSrc: `${BLOB_API_PATH}?folderName=photos&blobName=${file.name}` })
```

Reply:
yes it is possible, but for now it is just a stub, the final functionality will not look like this. I think whoever is going to do the blob will decide how best to do it